From 0ba36071e4c843a999dd1d49349869c6d6d11303 Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Fri, 5 Jul 2013 15:43:41 +0900
Subject: [PATCH 615/675] ARM: shmobile: r8a7778: add
 r8a7778_add_usb_host_device()

The USB host driver was registered in compulsorily
on current r8a7778 SoC code.
But it should be decided on platform, not SoC.
This patch adds r8a7778_add_usb_host_device() for it.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
---
 arch/arm/mach-shmobile/board-bockw.c          |  7 ++++++-
 arch/arm/mach-shmobile/include/mach/r8a7778.h |  1 +
 arch/arm/mach-shmobile/setup-r8a7778.c        | 13 ++++++-------
 3 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/arch/arm/mach-shmobile/board-bockw.c b/arch/arm/mach-shmobile/board-bockw.c
index e25a310..68748e0 100644
--- a/arch/arm/mach-shmobile/board-bockw.c
+++ b/arch/arm/mach-shmobile/board-bockw.c
@@ -598,6 +598,11 @@ static void __init bockw_init(void)
 	platform_add_devices(bockw_devices, ARRAY_SIZE(bockw_devices));
 }
 
+static void __init bockw_init_late(void)
+{
+	r8a7778_add_usb_host_device();
+}
+
 static const char *bockw_boards_compat_dt[] __initdata = {
 	"renesas,bockw",
 	NULL,
@@ -610,5 +615,5 @@ DT_MACHINE_START(BOCKW_DT, "bockw")
 	.init_machine	= bockw_init,
 	.timer		= &shmobile_timer,
 	.dt_compat	= bockw_boards_compat_dt,
-	.init_late      = r8a7778_init_late,
+	.init_late      = bockw_init_late,
 MACHINE_END
diff --git a/arch/arm/mach-shmobile/include/mach/r8a7778.h b/arch/arm/mach-shmobile/include/mach/r8a7778.h
index 026d244..be3f158 100644
--- a/arch/arm/mach-shmobile/include/mach/r8a7778.h
+++ b/arch/arm/mach-shmobile/include/mach/r8a7778.h
@@ -38,6 +38,7 @@ extern void r8a7778_add_rsnd_devices(struct rcar_snd_info *rsnd_info,
 				     struct asoc_simple_card_info *card_info,
 				     int card_info_nr);
 extern void r8a7778_add_usb_phy_device(struct rcar_phy_platform_data *pdata);
+extern void r8a7778_add_usb_host_device(void);
 extern void r8a7778_add_vin_device(int id,
 				   struct rcar_vin_platform_data *pdata);
 extern void r8a7778_add_du_device(struct rcar_du_platform_data *info);
diff --git a/arch/arm/mach-shmobile/setup-r8a7778.c b/arch/arm/mach-shmobile/setup-r8a7778.c
index 27a2d39..66bf989 100644
--- a/arch/arm/mach-shmobile/setup-r8a7778.c
+++ b/arch/arm/mach-shmobile/setup-r8a7778.c
@@ -208,6 +208,12 @@ static struct platform_device_info hci##_info __initdata = {	\
 USB_PLATFORM_INFO(ehci);
 USB_PLATFORM_INFO(ohci);
 
+void __init r8a7778_add_usb_host_device(void)
+{
+	platform_device_register_full(&ehci_info);
+	platform_device_register_full(&ohci_info);
+}
+
 /* Ether */
 static struct resource ether_resources[] __initdata = {
 	DEFINE_RES_MEM(0xfde00000, 0x400),
@@ -698,12 +704,6 @@ void __init r8a7778_add_standard_devices(void)
 			ARRAY_SIZE(r8a7778_devices));
 }
 
-void __init r8a7778_init_late(void)
-{
-	platform_device_register_full(&ehci_info);
-	platform_device_register_full(&ohci_info);
-}
-
 static struct renesas_intc_irqpin_config irqpin_platform_data __initdata = {
 	.irq_base = irq_pin(0), /* IRQ0 -> IRQ3 */
 	.sense_bitfield_width = 2,
@@ -858,7 +858,6 @@ DT_MACHINE_START(R8A7778_DT, "Generic R8A7778 (Flattened Device Tree)")
 	.init_machine	= r8a7778_add_standard_devices_dt,
 	.timer		= &shmobile_timer,
 	.dt_compat	= r8a7778_compat_dt,
-	.init_late      = r8a7778_init_late,
 MACHINE_END
 
 #endif /* CONFIG_USE_OF */
-- 
1.8.2

